collector/generic_collector.go

```diff
@@ -160,6 +172,12 @@ func (collector *GenericCollector) Collect(metrics map[string][]v1.MetricVal) (t
 		} else {
 			errorSlice = append(errorSlice, fmt.Errorf("No match found for regexp: %v for metric '%v' in config", metricConfig.Regex, metricConfig.Name))
 		}
+		if len(newMetrics) > collector.info.metricCountLimit {
```

Comment:
Is this possible given that we fail to consume configs that define more metrics than what the `limit` permits?